src/framework/MOM_axis.F90

```diff
@@ -0,0 +1,625 @@
+     case ('Cu') ; use_lath = .true. ; use_lonq = .true. ! x=EAST_FACE, y=CENTER
+     case ('Cv') ; use_latq = .true. ; use_lonh = .true. ! x=CENTER, y=NORTH_FACE
+     case ('1') ; ! x=0, y=0
+     case default
+        call MOM_error(FATAL, "MOM_axis:get_var_dimension_features "//&
+                        "Unrecognized hor_grid argument "//trim(grid_string_id))
+  end select
+end subroutine get_horizontal_grid_logic
+
+!> Define the time units for the input time value
+function get_time_units(time_value) result(time_units_out)
+   real, intent(in) :: time_value !< numerical time value in seconds
+                                  !! i.e., before dividing by 86400.
+   ! local
+   character(len=10) :: time_units ! time units
+   character(len=10) :: time_units_out ! time units trimmed
+   time_units = ''
+   time_units_out = ''
+   if (time_value < 0.0) then
+      time_units = "days" ! The default value.
+   elseif (mod(time_value,86400.0)==0.0) then
+      time_units = "days"
+   elseif ((time_value >= 0.99) .and. (time_value < 1.01)) then
+              time_units = "seconds"
+   elseif ((time_value >= 3599.0) .and. (time_value < 3601.0)) then
+              time_units = "hours"
+   elseif ((time_value >= 86399.0) .and. (time_value < 86401.0)) then
+              time_units = "days"
+   elseif ((time_value >= 3.0e7) .and. (time_value < 3.2e7)) then
+              time_units = "years"
+   else
+       write(time_units,'(es8.2," s")') time_value
+   endif
+   time_units_out = trim(time_units)
+end function get_time_units
+
+!> function to get the index of a time_value from a netCDF file
+function get_time_index(filename, time_to_find) result (time_index)
+  character(len=*) :: filename ! name of the file to read in
+  real, intent(in) :: time_to_find ! time value to search for in file
+  ! local
+  type(fmsNetcdfFile_t) :: fileobj ! netCDF file object returned by open_file
+  real, allocatable, dimension(:) :: file_times ! array of time values read from file
+  integer :: dim_unlim_size, i, time_index
+  character(len=nf90_max_name) :: dim_unlim_name ! name of the unlimited dimension in the file
+  logical :: file_open_success
+
+  time_index = 1
+  dim_unlim_size = 0
+  dim_unlim_name = ""
+  file_open_success = .false.
+
+  if (.not. check_if_open(fileobj)) &
+    !call MOM_error(FATAL, "get_time_index_nodd: netcdf file object must be open.")
+    file_open_success=fms2_open_file(fileobj, trim(filename), "read", is_restart=.false.)
+
+  call get_unlimited_dimension_name(fileobj, dim_unlim_name)
+  call get_dimension_size(fileObj, trim(dim_unlim_name), dim_unlim_size)
+  ! time index will be one more than the unlimited dimension size if the time_to_find is not in the file
+  if (dim_unlim_size .gt. 0) then
+    time_index = dim_unlim_size+1
+    allocate(file_times(dim_unlim_size))
+    call read_data(fileobj,trim(dim_unlim_name), file_times)
+
+    do i=1,dim_unlim_size
```

Comment:
It seems to me that get_time_index should include proper error handling for the case when the time being requested is outside of the range of times spanned by the time index of the input file.